### PR TITLE
Rework of Docs

### PR DESCRIPTION
**Pull request by ArcasCZ**
The docs are bit messy, some of the information is not actual and it's a bit chaotic.

The rework is needed to get it to some level, when people can easily spin their own instance of the bot and get an environment for development.

### CODE REVIEW

File: docs/production/containers.rst [hunk 1]
```diff
@@ -130,27 +164,39 @@ It will probably be neccesary to add the user to the Docker group (this will tak
...
 
-	sudo apt install podman podman-docker docker-compose
+.. codeblock:: 
 
-Start the Podman system service:
+	[experimental]
+	sparseVhd=true
 
-.. code-block:: bash
+If you don't have WSL2 installend, you must run following command in ``cmd`` as Administrator.
```
Comment (kasicka) [inline docs/production/containers.rst:191]: Typo in installed.
I'd use need to or have to instead of must.

**Review by kasicka — APPROVED** [review]

File: docs/development/installation.rst [hunk 2]
```diff
@@ -40,17 +40,37 @@ Under a colored button **Code**, you can obtain a SSH URL which will be used wit
...
+	apt install \
+	    python3 python3-dev python3-pip python3-venv python3-setuptools
 
-Besides ``git``, strawberry.py has additional system dependencies which have to be installed.
+For Windows and MacOS, Python can be installed from `Official site <https://www.python.org/downloads/>`_.
 
-.. include:: ../_snippets/_apt_dependencies.rst
+To simplify few things on Windows, we'd recommend to run the installer as administrator and select ``Customize instalation`` during setup to be able to install Python for all users.
 
+If you'd like to use multiple Python versions at once, we might recommend `pyenv <https://github.com/pyenv/pyenv>`_ for Unix based systems or `pyenv-win <https://github.com/pyenv-win/pyenv-win>` for Windows
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:73]: Removed might.

File: docs/development/installation.rst [hunk 3]
```diff
@@ -40,17 +40,37 @@ Under a colored button **Code**, you can obtain a SSH URL which will be used wit
...
 It may be on your system already.
 
+Please follow the official tutorial for `Installing GIT <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>`_ on your OS.
+
+.. _devel_system_pre_setup_python:
+
+Install Python
+------------
+
+The two most important things to install are Python and PIP (python package manager, on Windows and Mac should be part of Python installer).
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:60]: Re-written.

File: docs/development/installation.rst [hunk 4]
```diff
@@ -40,17 +40,37 @@ Under a colored button **Code**, you can obtain a SSH URL which will be used wit
 System setup
 ------------
 
-You'll need ``git``.
+There are few necessary things you have to install on your computer before you start with development.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:43]: Changed.

File: docs/development/developing_modules.rst [hunk 5]
```diff
@@ -1,14 +1,19 @@
...
 	Always start from ``main``, but never commit to ``main``.
+    You can also start from the fresh branch synchronized with ``main``.
 
 Let's say you want to make changes to our `Fun <https://github.com/strawberry-py/strawberry-fun>`_ repository.
 
-If you were to use **repository install** command, the bot would place it into the ``modules/`` directory.
-That's where you have to clone your fork as well, so the bot can find and load it.
+You have two options to install the forked repository. 
+Option one is using **repository install** command, which would place it into the ``modules/`` directory.
+Option two is cloning the repository manually using ``git`` commands.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/developing_modules.rst:15]: Added info.

File: docs/development/installation.rst [hunk 6]
```diff
@@ -72,61 +92,38 @@ Then you have to setup a link back to our main repository, which is usually call
...
-Bot environment
----------------
-
-See :ref:`general_venv` in chapter General Bot Information for instructions on how to setup a virtual environment.
-
-Once you are in virtual environment, you can install required libraries:
+We have two options with the database. We can either use PostgreSQL or SQLite.
 
-.. code-block:: bash
+When working with databases, it's always good for testing purposes to test out the new code on both of them, as soem features might not be available for one of them.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:103]: Fixed

File: docs/development/installation.rst [hunk 7]
```diff
@@ -72,61 +92,38 @@ Then you have to setup a link back to our main repository, which is usually call
...
 
-	python3 -m pip install wheel
-	python3 -m pip install -r requirements.txt
-	python3 -m pip install -r requirements-dev.txt
+For the start, we'd recommend using SQLite, because it's simple to access - the data file can be easily opened in tools like `SQLite Browser <https://sqlitebrowser.org/>`_.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:105]: Removed "For the start".

File: docs/development/installation.rst [hunk 8]
```diff
@@ -72,61 +92,38 @@ Then you have to setup a link back to our main repository, which is usually call
...
+The second argument for SQLite is also the fact that wiping the DB means just deleting ``strawberry.db`` file.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:106]: Changed

File: docs/development/installation.rst [hunk 9]
```diff
@@ -72,61 +92,38 @@ Then you have to setup a link back to our main repository, which is usually call
...
-.. code-block:: bash
+There are basically two recommended ways to run the bot.
 
-	python3 strawberry.py
+The recommended one is using docker as it makes the whole setup simple and the bot environment separated from your OS.
+This is specially useful when working with PostgreSQL. The setup is the same as described in :ref:`containers`.
+We can recommend using ``docker-compose up`` without ``detach`` parameter to see the bot's log easily.
 
-If you have done everything correctly (you are in ``venv``, you have all libraries installed), the script will print startup information and a welcome message, something like this:
+The second way is similar to :ref:`direct`. The only difference is that we can omit few things, such as PostgreSQL setup (and use SQLite only).
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:122]: Reworked based on your remarks.

File: docs/development/installation.rst [hunk 10]
```diff
@@ -144,3 +141,26 @@ If you have done everything correctly (you are in ``venv``, you have all librari
...
+.. note::
+	This is not necessary to test out the bot if you are using :ref:`containers` to run the bot.
+
+There are two reasons to install all the requirements. The first is that your IDE should be able to give you hints based on the installed python libraries.
+The second reason is that you can use pre-commit hook, which helps us maintain :ref:`code_quality`.
+
+Using virtual environment is optional and depends on your OS and IDE as some of them (e.g. PyCharm) have it's own way of managing virtual environments.
+See :ref:`direct_venv` in chapter Production section on how to setup a virtual environment.
+
+This code can be also run directly in your main (not virtual) environment, but in that case there might be colisions with already installed libraries.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst:159]: Added l

File: docs/development/repository.rst [hunk 11]
```diff
@@ -170,4 +170,6 @@ Please note that this may be changed in the future and some strawberry.py versio
 Load module
 -----------
 
-For import and load of the custom modules follow `User documentation <https://strawberry-py.github.io/docs/en/module-installation.html>`_. The user documentation expected that module is available as git repository and everything required in :ref:`how_to_create_repo` is fulfilled.
+The user documentation expected that module is available as git repository and everything required in :ref:`how_to_create_repo` is fulfilled.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/repository.rst:173]: changed `expected` to `expects`

File: docs/development/repository.rst [hunk 12]
```diff
@@ -170,4 +170,6 @@ Please note that this may be changed in the future and some strawberry.py versio
...
+
+Then you can load the module as described in :ref:`installing_module`
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/repository.rst:175]: Added . at the end

File: docs/general/index.rst [hunk 13]
```diff
@@ -77,79 +35,11 @@ You can invite the bot to your server by going to the **OAuth2/URL Generator pag
...
-A small tip
-^^^^^^^^^^^
-
-When working on the bot (debugging, development) it is easier if you speed up environment variable import.
-Open the activate script (the ``.venv/bin/activate`` file) and insert to the end of it:
+Installing modules
+-----------------
 
-.. include:: ../_snippets/_source_env.rst
+Installing modules is done through the ``repository`` command of the bot instance or by manually clonning the module as described in :ref:`developing-modules`.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/general/index.rst:43]: Removed N

File: docs/general/index.rst [hunk 14]
```diff
@@ -77,79 +35,11 @@ You can invite the bot to your server by going to the **OAuth2/URL Generator pag
...
 
-This way the variables will be set whenever you enter the virtual environment with the ``source .venv/bin/activate`` command, and you won't have to run the ``source .env`` manually.
+To get more info of the repository command, please refer to ``help repo`` command of the bot instance.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/general/index.rst:45]: Changed `of` to `about`

File: docs/production/containers.rst [hunk 15]
```diff
@@ -130,27 +164,39 @@ It will probably be neccesary to add the user to the Docker group (this will tak
...
+.. warning::
+	This tutorial covers the installation on Windows 10 Build 2004 or later and should be compatible
+	with Windows 11 as well.
 
-Podman Installation
--------------------
 
-.. note::
-	If you already installed docker you can skip this part
+In this tutorial we will be working with Docker Desktop, which is free for non-commercial usage.
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/production/containers.rst:176]: Added coma.

File: docs/production/containers.rst (hunk 1 above)
Comment (ArcasCZ) [reply docs/production/containers.rst:191]: Changed

File: docs/development/installation.rst [hunk 16]
```diff
@@ -40,17 +40,37 @@ Under a colored button **Code**, you can obtain a SSH URL which will be used wit
...
+Please follow the official tutorial for `Installing GIT <https://git-scm.com/book/en/v2/Getting-Started-Installing-Git>` on your OS.
+
+.. _devel_system_pre_setup_python:
+
+Install Python
+------------
+
+The two most important things to install are Python and PIP (python package manager, on Windows and Mac should be part of Python installer).
+
+For Linux, we'd recommend using your system package manager (e.h. `apt`) to install Python and it's extensions:
```
[the remark this answers is not in the document]
Comment (ArcasCZ) [reply docs/development/installation.rst]: Changed

**Pull request merged**